lib/src/user.cc

```diff
@@ -33,9 +33,17 @@ namespace libral {
   static result<bool> add_group_list(resource& rsrc,
                                      const char* user, gid_t group) {
     int ngroups = 0;
+#ifdef __APPLE__
```

Comment:
This is fine for making things work, but I'd prefer if there was a cmake test that checks whether `gid_t` is available and sets `HAVE_GID_T` accordingly, and then have something like this at the top of `user.cc`:

```c
#if HAVE_GID_T
typedef int gid_t;
#endif
```

Even without a cmake test (which might be a bit of a headache), I'd prefer putting something like this into `config.hpp.in`:
```c
#if __APPLE__
// no gid_t type
#else
#define HAVE_GID_T 1
#if
```
since that is easier to refine later on.

I just checked gnulib, and they take a slightly different tack: they define `GETGROUPS_T` according to whether `gid_t` is available or not - that's also a possibility rather than using a typedef and a `HAVE_GID_T`.

The overall point is: platform-specific checks should go into cmake or `config.hpp.in`, and code should only make decisions based on fairly fine-grained information about what is and is not available.